cmd/kubelet/app/server.go

```diff
@@ -474,6 +476,30 @@ func makeEventRecorder(kubeDeps *kubelet.Dependencies, nodeName types.NodeName)
 	}
 }
 
+func getReservedCPUs(machineInfo *cadvisorapi.MachineInfo, cpus string) cpuset.CPUSet {
+	emptyCPUSet := cpuset.NewCPUSet()
+	if cpus != "" {
+		topo, err := topology.Discover(machineInfo)
+		if err != nil {
+			klog.Error("Unable to discover CPU topology info")
+			return emptyCPUSet
+		}
+		reservedCPUSet, err := cpuset.Parse(cpus)
+		if err != nil {
+			klog.Error("Unable to parse reserved-cpus list")
+			return emptyCPUSet
+		}
+		allCPUSet := topo.CPUDetails.CPUs()
+		klog.Infof("Online CPUs: %s", allCPUSet.String())
+		if !reservedCPUSet.IsSubsetOf(allCPUSet) {
+			klog.Error("Invalid reserved-cpus list")
+			return emptyCPUSet
```

Comment:
Will this silently ignore the given cpuset if it isn't valid? Are we sure we don't want to fail hard if that it the case? I dont' think this will be caught by config validation tho, but maybe I am wrong?

Reply:
right, the cli config validation won't catch this - I'm not sure if we can get cadvisorapi.MachineInfo at CLI parsing time, otherwise do a thorough check and fail hard at CLI config validation will be better